src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -142,6 +138,34 @@ public override string ToString()
             return s;
         }
 
+        // Returns the indented inner exception's ToString() prefixed, but not suffixed, with a newline.
+        // If there is no inner exception, returns empty string.
+        private protected string InnerExceptionToString(Exception? inner)
+        {
+            if (inner == null)
+                return "";
+
+            /* Format something like this to clearly separate from the containing exception
+             ________________________________________________________________________________________________
```

Comment:
I believe that the ascii art part of the change has significant risk of breaking tools that analyze exception stacktraces, and it may be controversial (e.g. my personal preference would be a more compact format without the ascii art, but I am happy to be overruled.) 

If you would like to pursue it, I think it should have its own issue and PR. The issue should be marked as breaking change and the stakeholders should be included to comment on it:
- The community projects that are in the business of formatting and understanding exception stacktraces (e.g. Ben.Demystifier)
- Our diagnostic and TI teams

Reply:
> breaking tools that analyze exception stacktraces

e.g. some of our own corefx tests are failing because of this problem

Reply:
I really just wanted to start a discussion. Any suggestions for an alternative? Maybe the arrows? (Is there general agreement the existing format is hard to correlate?)

Reply:
I'll close this and open an issue to discuss that...